src/nvim/eval/funcs.c

```diff
@@ -2063,6 +2063,7 @@ static void f_expand(typval_T *argvars, typval_T *rettv, FunPtr fptr)
       if (result != NULL) {
         tv_list_append_string(rettv->vval.v_list, (const char *)result, -1);
       }
+      xfree(result);
```

Comment:
but result may be NULL here ? move it up one line

Reply:
In the C89 specification, free is defined as follows. So there's nothing wrong with this code.

> The free funciton causes the space pointed to by ptr to be deallocated, that is, made available for further allocation. If ptr is a null pointer, no action occurs. Otherwise, if the argument does not match a pointer earlier returned by the calloc, malloc, or realloc function, or if the space has been deallocated by a call to free or realloc, the behavior is undefined.

Reply:
Looking at xfree's doc recommands to use XFREE_CLEAR. Looking at this peculiar instance and https://github.com/neovim/neovim/pull/10046#, looks like it qualifies.